### ISSUE TITLE
dovecot connection lost

### ISSUE THREAD

**Issue by catcun** [mailcow/mailcow-dockerized#1475]
Hello,
I am not sure if this is related to #1440 but postfix is losing connections to dovecot and I have messages stuck on the queue and being re-delivered periodically serveral times until I delete the message from the queue.

Queue example:
```
# docker-compose exec postfix-mailcow mailq
-Queue ID-  --Size-- ----Arrival Time---- -Sender/Recipient-------
6F0E398FFF24   154213 Tue Jun 12 13:19:24  georgef@publicdomain.com
(lost connection with dovecot[fd4d:6169:6c63:6f77::f] while sending end of data -- message may be sent more than once)
                                         dir@mydomain.com
                                         pers@mydomain.com
                                         pers1@mydomain.com
                                         rs@mydomain.com
```

postfix logs at the time:
```disconnect from mailcowdockerized_dovecot-mailcow_1.mailcowdockerized_mailcow-network[172.22.1.12] ehlo=1 mail=1 rcpt=1 data=1 commands=4
6F0E398FFF24: to=<rs@mydomain.com>, relay=dovecot[fd4d:6169:6c63:6f77::f]:24, delay=7502, delays=7487/0.01/0.1/14, dsn=4.4.2, status=deferred (lost connection with dovecot[fd4d:6169:6c63:6f77::f] while sending end of data -- message may be sent more than once)
6F0E398FFF24: to=<pers@mydomain.com>, relay=dovecot[fd4d:6169:6c63:6f77::f]:24, delay=7502, delays=7487/0.01/0.1/14, dsn=4.4.2, status=deferred (lost connection with dovecot[fd4d:6169:6c63:6f77::f] while sending end of data -- message may be sent more than once)
6F0E398FFF24: to=<pers1@mydomain.com>, relay=dovecot[fd4d:6169:6c63:6f77::f]:24, delay=7502, delays=7487/0.01/0.1/14, dsn=4.4.2, status=deferred (lost connection with dovecot[fd4d:6169:6c63:6f77::f] while sending end of data -- message may be sent more than once)
6F0E398FFF24: to=<dir@mydomain.com>, relay=dovecot[fd4d:6169:6c63:6f77::f]:24, delay=7502, delays=7487/0.01/0.1/14, dsn=4.4.2, status=deferred (lost connection with dovecot[fd4d:6169:6c63:6f77::f] while sending end of data -- message may be sent more than once)
```

dovecot logs at the time:
```lmtp(dir@mydomain.com)<9833><cMntGZzlH1tpJgAAuR7xiA>: sieve: msgid=<93E13061-B27E-42B6-A9C9-D9D155A887F1@publicdomain.com>: stored mail into mailbox 'INBOX'
lmtp(9833): Connect from fd4d:6169:6c63:6f77::e
lmtp(pers@mydomain.com)<9689><gAP6E5zlH1vZJQAAuR7xiA:3>: Fatal: master: service(lmtp): child 9689 killed with signal 6 (core not dumped - https://dovecot.org/bugreport.html#coredumps - set /proc/sys/fs/suid_dumpable to 2)
lmtp(pers@mydomain.com)<9689><gAP6E5zlH1vZJQAAuR7xiA:3>: Error: Raw backtrace: /usr/local/lib/dovecot/libdovecot.so.0(+0xc9bf1) [0x7f5f01347bf1] -> /usr/local/lib/dovecot/libdovecot.so.0(+0xc9cbd) [0x7f5f01347cbd] -> /usr/local/lib/dovecot/libdovecot.so.0(i_fatal+0) [0x7f5f012b7391] -> /usr/local/lib/dovecot/libdovecot.so.0(o_stream_send_istream+0x228) [0x7f5f0136e928] -> /usr/local/lib/dovecot/libdovecot.so.0(smtp_client_command_send_more+0x1a7) [0x7f5f012c0f07] -> /usr/local/lib/dovecot/libdovecot.so.0(+0x488c6) [0x7f5f012c68c6] -> /usr/local/lib/dovecot/libdovecot.so.0(+0xf26e0) [0x7f5f013706e0] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_call_io+0x69) [0x7f5f0135fa49] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_handler_run_internal+0x109) [0x7f5f01361339] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_handler_run+0x52) [0x7f5f0135fb52] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_run+0x38) [0x7f5f0135fd68] -> /usr/local/lib/dovecot/libdovecot.so.0(smtp_submit_run+0xba) [0x7f5f012d1c9a] -> /usr/local/lib/dovecot/lib90_sieve_plugin.so(+0x2a24) [0x7f5efef0fa24] -> /usr/local/lib/dovecot/libdovecot-sieve.so.0(+0x501da) [0x7f5efeca71da] -> /usr/local/lib/dovecot/libdovecot-sieve.so.0(+0x40581) [0x7f5efec97581] -> /usr/local/lib/dovecot/libdovecot-sieve.so.0(sieve_result_execute+0x217) [0x7f5efec98a97] -> /usr/local/lib/dovecot/libdovecot-sieve.so.0(+0x52b6e) [0x7f5efeca9b6e] -> /usr/local/lib/dovecot/libdovecot-sieve.so.0(sieve_multiscript_run+0xa6) [0x7f5efecaa9c6] -> /usr/local/lib/dovecot/lib90_sieve_plugin.so(+0x3b66) [0x7f5efef10b66] -> /usr/local/lib/dovecot/libdovecot-lda.so.0(mail_deliver+0xf0) [0x7f5f0194ed50] -> dovecot/lmtp(lmtp_local_data+0x4f1) [0x55f3c8a3afe1] -> dovecot/lmtp(cmd_data_continue+0x233) [0x55f3c8a39cf3] -> /usr/local/lib/dovecot/libdovecot.so.0(+0x4b940) [0x7f5f012c9940] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_call_io+0x69) [0x7f5f0135fa49] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_handler_run_internal+0x109) [0x7f5f01361339] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_handler_run+0x52) [0x7f5f0135fb52] -> /usr/local/lib/dovecot/libdovecot.so.0(io_loop_run+0x38) [0x7f5f0135fd68] -> /usr/local/lib/dovecot/libdovecot.so.0(master_service_run+0x13) [0x7f5f012dac83]
lmtp(pers@mydomain.com)<9689><gAP6E5zlH1vZJQAAuR7xiA:3>: Panic: file ostream.c: line 394 (o_stream_send_istream): assertion failed: (!_outstream->finished)
lmtp(pers@mydomain.com)<9689><gAP6E5zlH1vZJQAAuR7xiA:3>: Warning: smtp-client: conn postfix:588 [5]: Received invalid EHLO response line: Unexpected character in EHLO keyword
lmtp(pers1@mydomain.com)<9689><gAP6E5zlH1vZJQAAuR7xiA:2>: sieve: msgid=<93E13061-B27E-42B6-A9C9-D9D155A887F1@publicdomain.com>: stored mail into mailbox 'INBOX'
```

Mailcow dockerized on CentOS 7
```
git show-branch
[master] After update on 2018-05-29_01_06_57
```

Can anyone help me diagnose this? Thanks  a lot

**Comment by andryyy** [comment]
This is fixed in Dovecots master, I think. There is also a serious bug in replication over tcp. 

Someone shouts out loud when 2.3.2 is available?

**Comment by catcun** [comment]
@andryyy 
Understood, lets wait.

**Comment by develth** [comment]
Had it today, too. 

Can someone confirm that it is fixed with https://github.com/mailcow/mailcow-dockerized/commit/055183257dddc3e7303dab3adeace99f5d5fd772 ?
I didn´t updated until now.

Thanks

**Comment by andryyy** [comment]
Are you on 2.3.0 @develth ?

**Comment by develth** [comment]
I´m on 2.3.1

**Comment by andryyy** [comment]
Ah, yes, it was fixed in 2.3.2, not 2.3.1.
